packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -1031,7 +1037,25 @@ export function diffHydratedProperties(
             warnForPropDifference(propKey, serverValue, expectedStyle);
           }
         }
-      } else if (isCustomComponentTag) {
+      } else if (
+        enableCustomElementPropertySupport &&
+        isCustomComponentTag &&
+        (propKey === 'offsetParent' ||
+          propKey === 'offsetTop' ||
+          propKey === 'offsetLeft' ||
+          propKey === 'offsetWidth' ||
+          propKey === 'offsetHeight' ||
+          propKey === 'isContentEditable')
+      ) {
```

Comment:
I think there are quite a few more properties that throw when assigned to, aren't there?
A quick test indicates you may want to add at least `outerText` and `outerHTML`.

Reply:
Thanks! I just pushed a commit to include `outerText` and `outerHTML`. Fortunately this is just a warning, so if it's not perfect we can always change it later.